PVCleanup

dwp1084 opened the pull request.

dwp1084 committed:
Split PVC into abstract and inherited classes

dwp1084 committed:
Created new target file for go kart

Commit:
Applied Formatting Changes During GitHub Build

mjmagee991 reviewed (changes requested):
Seeing it all laid out like this, I think it might be easier to maintain and understand what's going on if the DEV1 and KEV1N classes are completely independent of each other. As it is now, you'd have to jump between PreChargeBase and PreCharge to follow the flow of logic, which is going to make maintaining it a bit of a pain. Ultimately, though, it's up to you whether you want these classes to be separate or together, because you're the person who has to maintain it.

The one other argument I'll give for separating them out is that it will let us merge this branch without doing any additional testing because the main PreCharge code won't have changed. If you do split things up like this, we're going to have to do quite a bit of testing to make sure everything is still working.

No solution is ideal, but I'd like to hear your thoughts.

dwp1084 committed:
Separated two classes fully

dwp1084 committed:
build error :'(

mjmagee991 reviewed (approved):
Looks great!

Prosper-Omiponle committed:
Three lines were missing from KEV1Precharge.hpp:

    static constexpr uint8_t MIN_PACK_VOLTAGE = 70;

    static constexpr uint8_t CONST_R = 30;
    static constexpr float CONST_C = 0.014;

This was causing the build error. There are still other CLion Compiler errors but they seem to be non-detrimental to the build

aclowmclaughlin commented on targets/PreCharge/main.cpp:
```diff
@@ -28,15 +28,15 @@ struct CANInterruptParams {
+* will be passed to the EVT-core CAN interface which will in turn call this
+* function each time a new CAN message comes in.
+*
+* NOTE: For this sample, every non-extended (so 11 bit CAN IDs) will be
```
Should you have the "for this sample" notes in here?

aclowmclaughlin commented on targets/PreChargeKEV1N/main.cpp:
```diff
@@ -0,0 +1,209 @@
+///////////////////////////////////////////////////////////////////////////////
+// CANopen specific Callbacks. Need to be defined in some location
+///////////////////////////////////////////////////////////////////////////////
+extern "C" void CONodeFatalError(void) {
```
This stuff will be unnecessary eventually with Taylor's CanOpen changes, but I assume it's all good for now.

aclowmclaughlin reviewed (approved):
Looks good, I think!
